### ISSUE TITLE
Constant and repeated FPS drop while playing tModLoader

### ISSUE THREAD

**Issue by Daisy92x** [bluemagic123/tModLoader#3333]
### Version

Latest 1.4-stable

### OS

Windows

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

When I start to play on tModLoader, game is fluid and everything works properly. When I start to go around, approaching new biomes or getting closer to new enemies or even when using some tools or weapons, I experiment an huge and persistent FPS drop.

### Log File

[client.log](https://github.com/tModLoader/tModLoader/files/11020575/client.log)

### Steps to reproduce

1. I login to the game and after few minutes I start to experiment huge FPS Drops.
2. They seems to be related to the approach of new enemies, biomes, osts, animations and so on
3. The only way I have to bring game back to fluidity again is to use ALT+TAB and switch windows.
4. Few minutes after using the ALT+TAB command, the issue shows up again.
5. The issue becomes gradually more frequent.

### Reproduction frequency

The issue is shown many times per minute, making he game unplayable.

### Additional Information

I tried to play on 1.3 legacy version and I don't have issues there. 
I tried to play on Vanilla and I don't have issues there.
I tried to play tModLoader both in windowed and full screen mode, tried all possible resolution but no one works.
I experience this issue both in single player and in multi player mode.
I tried to deactivate Autosave function and Frame Skip function but id didn't work.
I tried to set tModLoader processes on "high priority" in Task Manager details but it didn't work.
I deactivated Steam overlay for tModLoader but it didn't work.

**Comment by JavidPack** [comment]
There is not much to go off in your logs. The only thing I can see is you are using Calamity, which is a particularly complex mod, so it might be related to some specific logic in that mod.

If you experience this with other sets of mods, that would be helpful to know, but as it is now there isn't anything to go off of here, in terms of us diagnosing the issue.

**Comment by Daisy92x** [comment]
@JavidPack Thanks for your reply
So, I just tried to disable every mod active on my tModLoader and waited for the client to unload them all. After this, I created totally new character and world: nothing to do, after 1 minute the issue showed up again :\

Btw, I noticed that when I run tModLoader through Steam, it also starts to run Terraria (Vanilla) together with it (no Vanilla client physically opened, but Steam just says they are both running)... is it normal? 

Anyway... is there any other thing I could set/change to try to solve it? :(
Uninstalling and reinstalling it could help? 
[client.log](https://github.com/tModLoader/tModLoader/files/11022207/client.log)

**Comment by cozykoko** [comment]
Them both running is normal. What are your pc specs?

**Comment by Daisy92x** [comment]
@cozykoko thanks for your reply

My pc specs are:
CPU: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz - Number of Cores : 8 - Speed : 4 GHz
GPU: NVIDIA GeForce MX130
RAM: 12 GB
OS: Windows 10 64bit

Not the best of pc ofc, but I play games way more heavy than tModLoader (graphically speaking) on it and I've no fps issues there. 
Besides, my pc runs Terraria and tModLoader 1.3 legacy perfectly, no fps drops there.
To sum it up, I encounter this issue only on tModLoader 1.4 (stable)

**Comment by JavidPack** [comment]
Hm, well, we certainly will keep an eye out. There isn't anything specific in your logs that would suggest any issue, so we'll have to see if we notice anything as we continue to work on tmodloader.

One thing I did notice is that you disabled the mods and then tested with 0 mods in the same game session. This should work in theory, but there is a chance something is still changed from an unloaded mod. I would suggest doing a test of closing the game after disabling all mods, then opening the game again with 0 mods enabled. Confirm for us that it still happens in that situation, and send the new client.log as well.

**Comment by Daisy92x** [comment]
@JavidPack thanks for your reply
I made the test you asked for and I'm still experiencing the issue, sadly. (Here is the new client log)

Any suggestion about any setting I could change (both on tModLoader and PC) which could help me solve this issue? :(
[client.log](https://github.com/tModLoader/tModLoader/files/11039266/client.log)

**Comment by Chicken-Bones** [comment]
If you're tech savvy enough, you could use a free 30 day trial of https://www.jetbrains.com/profiler/ to record a timeline and send it to us. Otherwise there's not much we can do except keep an eye out

**Comment by JavidPack** [comment]
This issue has either no recent activity or no reproducible steps, so it is being closed. We assume this issue has solved itself in some manner. Feel free to open a new issue if the issue persists and fill out the issue template completely.

**Comment by Falafelwaffle07** [comment]
Hey I'm having the same issue, granted I usually run multiple mods Tmodloader is stuttering every 10 steps it feels

**Comment by Falafelwaffle07** [comment]
Device name	DESKTOP-PALS4UG
Processor	Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz   2.81 GHz
Installed RAM	16.0 GB (15.9 GB usable)
Device ID	837C8C00-4DA0-4484-8567-976AE4CC25B4
Product ID	00326-10000-00000-AA935
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

those are my specs

**Comment by solderrw** [comment]
My friend has the same problem, it is the computer itself that has a 25% processor and the game itself is lagging. Can you help with this somehow?
